services/frontend/src/utils/models/Profile.js

```diff
@@ -103,4 +105,58 @@ export default class Profile {
 
         return profilesWithSkills;
     }
+
+    static addSkills = (profile, updatedSkills, skills) => {
+        const currentSkillsLC = profile.skills.map((skill) => skill.name.toLowerCase());
+
+        /* Returns a list of the new skills entered */
+        const newSkills = removeDuplicate(updatedSkills.reduce((acc, skillName) => {
+            if (!currentSkillsLC.includes(skillName.toLowerCase())) {
+                return [...acc, skillName];
+            }
+            return acc;
+        }, []));
+
+        /* Adds skill from database or Create new Skill */
+        if (newSkills.length > 0) {
+            for (const skill of Object.values(skills)) {
+                const skillName = skill.name;
+                const skillNameLC = skill.name.toLowerCase();
+                if (newSkills.includes(skillNameLC)) {
+                    profile.skills.push(skill);
+                    newSkills.splice(newSkills.indexOf(skillName), 1);
+                }
+            }
+        }
+
+        /* Adds new Skill */
+        if (newSkills.length > 0) {
+            for (const skillName of newSkills) {
+                const newSkill = new Skill({name: skillName});
+                profile.skills.push(newSkill);
+            }
+        }
+        return profile;
+    };
+
+    static removeSkills = (profile, updatedSkills) => {
```

Comment:
This is currently bugged with the old for loop plus splicing problem. I assume you have encountered it before but if not, try to delete all of skills at the same time and it will only delete two since as you go through the loop you are making the array shorter and therefore messing with the indexes.

Reply:
As this is said to be part of another ticket I will resolve this for now.